modules/ROOT/pages/dataweave-formats.adoc

```diff
@@ -469,13 +469,109 @@ contains header field names. . Valid options: `true` or `false`
 | `separator` | Char | `,` a| Character that separates one record from another.
 | `streaming` | Boolean | `false` a| Used for streaming input CSV.
 Valid options: `true` or `false` (Use only if entries are accessed sequentially.)
+See the <<stream_csv, streaming example>>, and see
+xref:dataweave-language-introduction.adoc#dw_readers_writers[DataWeave Readers and Writers].
 |===
 
 * When `header=true` you can then access the fields within the input anywhere
 by name, for example: `payload.userName`.
 * When `header=false` you must access the fields by index, referencing first
 the entry and then the field, for example: `payload[107][2]`
 
+[[stream_csv]]
+.Streaming Example:
+
+By default, the CSV reader stores input data from an entire file in-memory as
```

Comment:
CSV reader doesn't keep all the input in memory, if the input is larger than the set threshold (1.5 MB) it writes it to disk. Then the content is indexed and accessed on demand from the backing store (either disk or memory). But entries are kept in memory on demand, not the whole document at once, so we can handle big files even without streaming set to true.
Streaming is still beneficial for performance reasons though.

Reply:
added info